pluginconfig/commit.go

```diff
@@ -44,20 +47,37 @@ type TokenInfo struct {
 
 	// Decimals is the number of decimals for the token (NOT the feed).
 	Decimals uint8 `json:"decimals"`
+
+	// ChainFamily is the chain family the token is deployed on.
+	ChainFamily string `json:"chainFamily"`
 }
 
 func (a TokenInfo) Validate() error {
 	if a.AggregatorAddress == "" {
 		return errors.New("aggregatorAddress not set")
 	}
 
-	// aggregator must be an ethereum address
-	decoded, err := hex.DecodeString(strings.ToLower(strings.TrimPrefix(string(a.AggregatorAddress), "0x")))
-	if err != nil {
-		return fmt.Errorf("aggregatorAddress must be a valid ethereum address (i.e hex encoded 20 bytes): %w", err)
-	}
-	if len(decoded) != 20 {
-		return fmt.Errorf("aggregatorAddress must be a valid ethereum address, got %d bytes expected 20", len(decoded))
+	switch a.ChainFamily {
+	case chain_selectors.FamilySolana:
+		decoded, err := base58.Decode(string(a.AggregatorAddress))
+		if err != nil {
+			return fmt.Errorf("aggregatorAddress must be a valid Solana address (i.e. base58 encoded): %w", err)
+		}
+		if len(decoded) != 32 {
+			return fmt.Errorf("aggregatorAddress must be a valid Solana address, got %d bytes expected 32", len(decoded))
+		}
```

Comment:
Aggregators are only ever deployed on arbitrum and we'll hopefully be switching to Mercury some point thereafter or use keystone feeds. Don't think this makes sense but let me know the use case and we can see what you need.